packages/plugin-react/package.json

```diff
@@ -19,16 +28,18 @@
   "repository": {
     "type": "git",
     "url": "git+https://github.com/vitejs/vite.git",
-    "directory": "packages/plugin-react-refresh"
+    "directory": "packages/plugin-react"
   },
   "bugs": {
     "url": "https://github.com/vitejs/vite/issues"
   },
-  "homepage": "https://github.com/vitejs/vite/tree/main/packages/plugin-react-refresh#readme",
+  "homepage": "https://github.com/vitejs/vite/tree/main/packages/plugin-react#readme",
   "dependencies": {
     "@babel/core": "^7.12.13",
+    "@babel/plugin-transform-react-jsx": "^7.14.3",
     "@babel/plugin-transform-react-jsx-self": "^7.12.13",
     "@babel/plugin-transform-react-jsx-source": "^7.12.13",
-    "react-refresh": "^0.9.0"
+    "react-refresh": "^0.9.0",
+    "vite-react-jsx": "^1.1.0"
```

Comment:
Do you think we should inline it as the part of the package?

Reply:
Ideally, we should, but we need someone to either (a) migrate the new `plugin-react` to TypeScript or (b) migrate the `vite-react-jsx` code to JavaScript.

Reply:
Think we could migrate it into typescript if necessary.

Reply:
Migrate to TypeScript? I heard my name?! :eyes: 
What should I do? :laughing:

Reply:
Just need tsconfig.json, and have esbuild (in `build` script) use the ts loader, and add missing type annotations.

Reply:
We can take `plugin-vue` as a reference for TS setup I think. @aleclarson @Shinigami92 maybe you two can work on this PR together?

Reply:
and update the `exports` field in package.json

Reply:
I started with the rewrite here: https://github.com/aleclarson/vite/pull/1